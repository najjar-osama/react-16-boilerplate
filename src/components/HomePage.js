import React from "react";
import { connect } from "react-redux";
import { history } from "../router/AppRouter";
import { logout } from "../actions/auth";
export const HomePage = props => (
  <div>
    <h1>Home Page!!</h1>
    <button
      onClick={() => {
        props.logout();
        history.push("/");
      }}
    >
      Logout
    </button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(logout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(HomePage);
