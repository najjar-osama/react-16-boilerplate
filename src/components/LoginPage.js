import React from "react";
import { history } from "../router/AppRouter";
import { connect } from "react-redux";
import { login } from "../actions/auth";
export const LoginPage = props => (
  <div>
    <h1>Login Page</h1>
    <button
      onClick={() => {
        props.login("123abc");
        history.push("/home");
      }}
    >
      Login
    </button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  login: uid => dispatch(login(uid))
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
