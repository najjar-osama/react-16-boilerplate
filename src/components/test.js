import React from "react";

export const test = () => (
  <div>
    <h1>Test</h1>
    <h2>Hahaha</h2>
    <h3>jajas</h3>
  </div>
);
