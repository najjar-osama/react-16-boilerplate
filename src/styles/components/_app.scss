.main {
  align-items: center;
  display: flex;
  justify-content: center;
  min-height: 100vh;
  text-align: center;
  width: 100%;
}

.title {
  display: block;
}
.subtitle {
  display: block;
}
