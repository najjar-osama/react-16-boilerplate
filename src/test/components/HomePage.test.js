import React from "react";
import { shallow } from "enzyme";
import { HomePage } from "../../components/HomePage";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<HomePage />);
});

test("It should render HomePage component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
