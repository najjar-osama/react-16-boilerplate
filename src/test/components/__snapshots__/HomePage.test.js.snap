// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render HomePage component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Home Page!!
</h1>,
        <button
          onClick={[Function]}
>
          Logout
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home Page!!",
        },
        "ref": null,
        "rendered": "Home Page!!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Home Page!!
</h1>,
          <button
            onClick={[Function]}
>
            Logout
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home Page!!",
          },
          "ref": null,
          "rendered": "Home Page!!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
